packages/emotion-primitives/src/emotion-native.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import transform from 'css-to-react-native'
+import cssToObjects from 'css-to-object'
+import primitives from 'react-primitives'
+
+import { getStyles, resolveStyles } from './resolveStyles'
+import { splitProps, getPrimitiveProps } from './splitProps'
+
+const StyleSheet = primitives.StyleSheet
+
+/*
+  Renders the styles on iOS, Android and Sketch. Styles are first converted to objects, then to a valid format using StyleSheet.create
+
+  TODO: Pass props to interpolate function
+*/
+function renderStylesOnNative(tag, styles) {
+  // Raw css styles
+  let css
+
+  // Store styles which will be converted to RN
+  const cssToRN = []
+
+  // Store RN styles
+  let rnStyles
+
+  // If styles are not passed using the object styles
+  if (typeof styles === 'object' && Object.keys(styles) == 0) {
+    css = cssToObjects(styles.join(''))
+
+    Object.keys(css).forEach(prop => {
+      cssToRN.push([prop, css[prop]])
```

Comment:
`cssToRN` is only manipulated here and passed as argument to a function just few lines below, as it's size is known (`Object.keys(css).length`) it might be worth to preallocate it based on that

I mean smth like this:
```js
const keys = Object.keys(css)
const cssToRN = new Array(keys.length)
keys.forEach((prop, i) => {
  cssToRN[i] = [prop, css[prop]]
})
```

Reply:
Hm! Makes sense. But what about `Array` instantiation cost ?